### PR TITLE
simple refactor of turbo shim

### PR DESCRIPTION
**Pull request by johrstrom**
This is a simple refactor of turbo_shim.js to promote reuse because basically every component is going to poll as well as replace.

### CODE REVIEW

File: apps/dashboard/app/views/batch_connect/sessions/index.html.erb [hunk 1]
```diff
@@ -36,7 +36,7 @@ locals: {
     </div>
     <div class="col-md-9">
       <div id="batch_connect_sessions">
-        <%= render(partial: 'panels', locals: { use_template: false }) %>
+        <%= render(partial: 'panels') %>
```
Comment (johrstrom) [inline apps/dashboard/app/views/batch_connect/sessions/index.html.erb:39]: Not really a part of this js refactor, just happened to notice that this is unused.

File: apps/dashboard/app/javascript/turbo_shim.js [hunk 2]
```diff
@@ -19,3 +19,15 @@ export function replaceHTML(id, html) {
...
 }
+
+export function pollAndReplace(url, delay, id) {
+  fetch(url, { headers: { Accept: "text/vnd.turbo-stream.html" } })
+    .then(response => response.ok ? Promise.resolve(response) : Promise.reject(response.text()))
+    .then((r) => r.text())
+    .then((html) => replaceHTML(id, html))
+    .then(setTimeout(pollAndReplace, delay, url, delay, id))
+    .catch((err) => {
+      console.log('Cannot not retrive partial due to error:');
```
Comment (Oglopf) [inline apps/dashboard/app/javascript/turbo_shim.js:30]: This all looks good, but it's early, and I swear this is reading as a double negative (but I'm brewing my coffee still). Should this just be `Cannot retrieve partial due to error:` or am I still waking up?

Comment (johrstrom) [reply apps/dashboard/app/javascript/turbo_shim.js:30]: Lol yea, that's a double negative. I'll update now.

**Review by Oglopf — APPROVED** [review]
Looks good!

**Pull request merged**